src/debugpy/adapter/clients.py

```diff
@@ -511,6 +515,11 @@ def configurationDone_request(self, request):
             if conn.server is None and conn.ppid == self.session.pid:
                 self.notify_of_subprocess(conn)
 
+    @message_handler
+    def evaluate_request(self, request):
+        return self.server.channel.delegate_async(request)
+        return NO_RESPONSE
```

Comment:
I think `OutgoingRequest.on_response()` should be able to solve this without a helper to spawn additional threads etc. It'll invoke the callback on a background thread dedicated to message handlers, with response as the argument. 

So, something like:
```py
    @message_handler
    def evaluate_request(self, request):
        propagated_request = self.propagate(request)

        @propagated_request.on_response
        def handle_response(response):
             
             request.respond(response.body)  # this will also do the right thing if body is Exception
               
        return messaging.NO_RESPONSE
```
(The callback is also guaranteed to be called if the server disconnects before responding - `response.body` will be `messaging.NoMoreMessages` in that case, which should still work fine with the above code.)